Git-ignore developer-specific files

michaelbaudino opened the pull request.

tjouan commented on README.md:
```diff
@@ -14,6 +14,16 @@ Code style used by AlpineLab in all our projects.
+1. Developer-environment-specific files must be `.gitignore`'d, _e.g._:
+
+    * `config/database.yml`
+    * `.ruby-{version,gemset}`
```
I think that those are two different cases:
* `config/database.yml` it's more related to the specific project, than the
  developer environment, it also affects direct "users", not only developers;
* `.ruby-{version,gemset}` should not be in the git ignore rules of a
  repository, they should be either in the global git ignore rules, or in local
  config specific to a repository clone.

michaelbaudino committed:
Add best practices to README

Those best practices only include some configuration files to
git-ignore, for now.

Polo2 reviewed (approved):
Understood, and agreed
